BOLD-03.md

```diff
@@ -1,114 +1,77 @@
+* The **Maker**: propagating an order to its peers when it couldn't find a match in the current order book. The order then joins a pool of unmatched orders on all connected peers. 
+* The **Taker**: issuing an order which immediately can be matched with an existing (maker) order in the current order book.
 
-	**string** id = 1
+In order to sustain the robustness and censorship resistance property of the network, OpenDEX lacks a global order book and a centralized matching engine. Instead, each node on the network maintain its own order book and matching engine. The order book can be limited to a preferred set of trading pairs. In case of a match, the taker order node is expected to instantiate the swap protocol.
 
-		Message unique identifier, generated by the sender 
+###The Order Message (0x06)
 
-	**Order** order = 2
+    1. string id 
+	Message's globally unique identifier, generated by the sender 
 
-		The carried order 
+	2. Order order
+	The carried order 
 
-This message can be sent in order to announce the peer about a new open order.
+This message is used to announce a peer about a new maker order. It should be used only after the order could not be matched locally. 
 
-...
+###The Order Type
 
-**The Order Type**
+	1. string id
+	Order's globally unique identifier, generated by the sender 
 
-	**string** id = 1
+    2. string pair_id
+    A trading pair symbol, constructed with the base currency first, followed by a '/' separator and the quote currency (e.g., “BTC\LTC”)
 
-		Order unique identifier, generated by the sender 
+    3. double price
+    The price for the order expressed in units of the quote currency
 
+    4. uint64 quantity
+    The number of currently satoshis (or equivalent) for the order
 
-    **string** pair_id = 2
+    5. bool is_buy
+    Whether the order is a buy (true) or a sell (false)
 
+###The OrderInvalidation Message (0x07)
 
-        A trading pair symbol, constructed with the base currency first, followed by a '/' separator and the quote currency (e.g., “BTC\LTC”)
+	1. string id
+	Message's globally unique identifier, generated by the sender 
 
+    2. string order_id
+    The associated order unique identifier
 
-    **double** price = 3
+    3. string pair_id
+    The trading pair symbol associated with the order
 
+    4. uint64 quantity
+    The number of satoshis (or equivalent) to invalidate from the order sum
 
-        The price for the order expressed in units of the quote currency
+This message is used to announce a peer the invalidation of an existing order quantity. It will allow the peer to remove the order from his local order book. Order invalidation is a usual event which occurs due to order cancellation or filling (by another order). Failing to announce it will result in the peer having a stale order in his local order book, which he might fill and instantiate a swap procedure which is doomed to fail. In this case, the maker order node reputation is be penalized. 
```

Comment:
"This message is used to tell a peer about the invalidation of an existing order or partial order."